swag/bank.py

```diff
@@ -301,6 +308,226 @@ def set_timezone(self, user, timezone):
 
         return lock_date
 
+    ## Ajout des fonctions des cagnottes
+
+    def create_cagnotte(self, cagnotte_name: str, currency: str, creator_id: int):
+        self.swagdb.add_cagnotte(cagnotte_name, currency, creator_id)
+        self.transactional_save()
+
+    def get_active_cagnotte(self, cagnotte_idx) -> Cagnotte:
+        if cagnotte_idx >= 0 and cagnotte_idx < self.swagdb.cagnotte_number():
+            cagnotte = self.swagdb.get_cagnotte_from_index(cagnotte_idx)
+            if cagnotte.cagnotte_activation == True:
+                return cagnotte
+
+        raise NoCagnotteRegistered(cagnotte_idx)
+
+    def get_all_active_cagnotte(self):
+        return [
+            cagnotte
+            for cagnotte in self.swagdb.get_cagnotte_infos()
+            if cagnotte.cagnotte_activation == True
+        ]
+
+    def get_cagnotte_history(self, cagnotte_idx):
+        return [
+            transaction
+            for transaction in self.swagdb.blockchain
+            if concerns_cagnotte(cagnotte_idx, transaction)
+        ]
+
+    def payer_a_cagnotte(
+        self, donator_account_discord_id: int, cagnotte_idx: int, montant
+    ):
+        cagnotte = self.get_active_cagnotte(cagnotte_idx)
+        donator_account = self.swagdb.get_account(donator_account_discord_id)
+        # On regarde le type de la cagnotte pour pouvoir correctement choisir les fonctions qui devront être utiliser
+        if cagnotte.get_info().cagnotte_currency == "$wag":
```

Comment:
I sought, and it's seems we never create a Currency Enum. 

I think it will not be necessary, if we create the $wag class and the $tyle class as you mentioned here : https://github.com/BobbySoftwares/Bybbot/pull/7#discussion_r675909749